.github/workflows/build-llvm-image.yml

```diff
@@ -4,67 +4,68 @@
 name: Build x86 LLVM-Nightly Image
 
 on: 
-  # Allow manually triggering rebuild.
+  pull_request:
   workflow_dispatch:
   # Run every week (on Sunday). 
   schedule:
     - cron:  '0 0 * * SUN'
 
-env:
-  REGISTRY: ghcr.io
-  IMAGE_NAME: ${{ github.repository }}
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
 
 jobs:
-  build-llvm-image:
+  buildandtest:
+    name: Build & Test
     runs-on: ubuntu-latest
-    steps:
-    # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-    - name: Get Repo
-      uses: actions/checkout@v2
-      with:
-        fetch-depth: 2
-        submodules: "true"
-        
-    - name: Ccache for C++ compilation
-      uses: hendrikmuhs/ccache-action@4687d037e4d7cf725512d9b819137a3af34d39b3
-      with:
-        key: ${{ runner.os }}-clangreleaseasserts-${{ steps.get-submodule-hash.outputs.hash }}
 
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v1
-        
-    - name: Log in to the Container registry
-      uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+    steps:
+    - uses: actions/checkout@v4
       with:
-        registry: ${{ env.REGISTRY }}
-        username: ${{ github.actor }}
-        password: ${{ secrets.GITHUB_TOKEN }}
+        submodules: true
 
-    - name: Extract metadata (tags, labels) for Docker
-      id: meta-llvm
-      uses: docker/metadata-action@98669ae865ea3cffbcbaa878cf57c20bbf1c6c38
+    - name: Setup ccache
+      uses: hendrikmuhs/ccache-action@v1
```

Comment:
Could we also use `actions/cache` here? What are the benefits of `hendrikmuhs/ccache-action`?

Reply:
I think those are two different types of caching. Note that the first is called `cache` and the second is the `ccache`. The latter I got it from the upstream llvm workflows and it installs and configures `(s)ccache`. For looking at other workflow files, the first one seems to be a normal cache that adds persistent storage between runs and that can be used more generically

Reply:
Ok, I was just wondering because it is a less-standard action to use and suspected it could break more easily. But if LLVM uses it also, I guess it will be maintained for a while.